docs/topics/spiders.rst

```diff
@@ -369,70 +369,37 @@ See `Scrapyd documentation`_.
 Start requests
 ==============
 
-Scrapy does not try to send :meth:`~scrapy.Spider.start` requests in order.
-Instead, it prioritizes reaching :setting:`CONCURRENT_REQUESTS` and
-:ref:`scheduling <topics-scheduler>` start requests.
-
-..
-    The request send order when all start requests and callback requests have
-    the same priority is rather unintuitive:
-
-    1.  First, the first CONCURRENT_REQUESTS start requests are sent in order.
-
-        Awaiting slow operations in Spider.start() can lower that.
-
-    2.  Then, assuming an even domain distribution in start requests (i.e.
-        ABCABC, not AABBCC), the last N start requests are sent in reverse
-        order, where N is:
-
-            min(CONCURRENT_REQUESTS, CONCURRENT_REQUESTS_PER_DOMAIN * domain_count)
-
-    3.  Finally, the remaining start requests are also sent in reverse order,
-        but only when there are not enough pending requests yielded from
-        callbacks to reach the configured concurrency.
-
-    The reverse order is because the scheduler uses a LIFO queue by default
-    (SCHEDULER_MEMORY_QUEUE, SCHEDULER_DISK_QUEUE). The order of the first few
-    requests is unnaffected because they are sent as soon as they are
-    scheduled. The last start requests sent before callback requests are those
-    that can be sent before the first callback requests are scheduled.
-
-    We do not document this behavior, so that we may change it in the future
-    without breaking the contract.
+**Start requests** are :class:`~scrapy.Request` objects yielded from the
+:meth:`~scrapy.Spider.start` method of a spider or from the
+:meth:`~scrapy.spidermiddlewares.SpiderMiddleware.process_start` method of a
+:ref:`spider middleware <topics-spider-middleware>`.
 
 .. _start-requests-order:
 
-Forcing a start request order
------------------------------
+Start request priority and order
+--------------------------------
+
+By default, :meth:`~scrapy.Spider.start` requests are handled differently from
+other requests:
 
-To force a specific **request order**, override the
-:meth:`~scrapy.Spider.start` method to set :attr:`Request.priority
-<scrapy.http.Request.priority>`. For example:
+-   Given the same :attr:`~scrapy.http.Request.priority`, start requests take
+    precedence.
 
--   To send start requests before other requests:
+-   Start requests are sent in the order they are yielded from
+    :meth:`~scrapy.Spider.start`, whereas other requests are sent in LIFO_
+    order.
 
-    .. code-block:: python
+    .. _LIFO: https://en.wikipedia.org/wiki/Stack_(abstract_data_type)
```

Comment:
What do you think about linking to https://docs.scrapy.org/en/latest/faq.html#does-scrapy-crawl-in-breadth-first-or-depth-first-order? It seems it also makes sense to expand this docs section to mention start priorities. Or maybe even combine these two sections, not sure.

Reply:
A nitpick: I'm asking mostly because of the next sentence, "you can change this in different ways", where it's not 100% clear what "this" is - it seems the intention was to explain how to customize start requests, but the last mention was of other requests. Anyways, it seems the explanation is useful for regular requests as well, and it partially overlaps with the FAQ entry.

Reply:
The new docs look good!